docs/ucai/index.md

```diff
@@ -0,0 +1,253 @@
+    - Solution: Verify that your specified warehouse has serverless compute enabled. Refer to the Databricks Serverless Compute Guide for setup instructions.
+
+- **Databricks SDK Installation**
+
+    - Pitfall: Missing or outdated databricks-sdk can prevent the client from functioning correctly.
+    - Solution: Ensure that you have installed the databricks-sdk as specified in the Prerequisites section. Keep it updated to the latest version to benefit from recent fixes and features.
+
+- **Profile Configuration**
+
+    - Pitfall: Incorrect profile settings can result in connection issues.
+    - Solution: If you're using multiple profiles, specify the correct one during client initialization. Ensure that the profile has the necessary permissions to interact with the Unity Catalog.
+
+- **Dependency Management**
+
+    - Pitfall: Missing dependencies like pandas can cause runtime errors when executing functions that require them.
+    - Solution: Ensure all necessary dependencies are installed in the environment where the client operates. Use virtual environments to manage dependencies cleanly.
+
+- **Compatibility with Databricks Connect**
+
+    - Pitfall: Using incompatible versions of databricks-connect can lead to execution failures, especially with serverless compute.
+    - Solution: Install the specific version of databricks-connect that is supported (15.1.0 or higher) to ensure compatibility with serverless features.
+
+
+#### Example Usage of the Client
+
+``` python
+from ucai.core.databricks import DatabricksFunctionClient
+
+# Initialize the client using serverless
+client = DatabricksFunctionClient()
+
+# Define your catalog and schema
+CATALOG = "my_catalog"
+SCHEMA = "my_schema"
+
+# Verify client setup by listing existing functions
+functions = client.list_functions(catalog=CATALOG, schema=SCHEMA)
+for func in functions.items:
+    print(func.full_name)
+```
+
+## Creating Functions
+
+### Using a Python Callable
+
+#### How the `create_python_function` API Works
+
+The `create_python_function` API in Unity Catalog AI provides a way to create Unity Catalog functions (UDFs) from Python functions, while ensuring compatibility with SQL environments. Here’s a comprehensive guide explaining how it works, what is required, and what is not allowed when using this API.
+
+##### Key Concepts
+
+- **Type Hint Requirements**
+    - All Python functions passed to `create_python_function` must have explicit type hints for both parameters and return values. These type hints are used to determine the corresponding SQL data types for the UC function.
+    - If type hints are missing, the function will not be accepted.
+   
+- **Supported Python to SQL Type Mapping**
+    - `int` → `LONG`
+    - `float` → `DOUBLE`
+    - `str` → `STRING`
+    - `bool` → `BOOLEAN`
+    - `Decimal` → `DECIMAL(38,18)` (with default precision and scale)
```

Comment:
I thought default is Decimal(10, 0) https://docs.databricks.com/en/sql/language-manual/data-types/decimal-type.html#syntax

Reply:
ah I forgot about that. We can't use the default though since it's lossy for large values. I map to the max scale to avoid truncation / rounding issues.